Source/SplineFollow/Private/Components/SFSplineComponent.cpp

```diff
@@ -77,88 +77,35 @@ void USFSplineComponent::UpdateSpline()
-            Cast< USFSplineMarkerObject_LevelActor >( new_marker.Object )->LevelActor = level_actor_marker.LevelActor;
-            SplineMarkers.Emplace( new_marker );
-        }
-
-        LevelActorActionMarkers.Empty();
-
-        for ( const auto & data_marker : DataMarkers )
-        {
-            auto new_marker = create_marker( data_marker, NewObject< USFSplineMarkerObject_Data >( this ) );
-            Cast< USFSplineMarkerObject_Data >( new_marker.Object )->Data = data_marker.Data;
-            SplineMarkers.Emplace( new_marker );
-        }
-
-        DataMarkers.Empty();
-    }
-
-    Super::Serialize( archive );
+    return FSFSplineMarker(
+        default_marker.Name,
+        default_marker.ItIsEnabled,
+        default_marker.Infos,
+        NewObject< USFSplineMarkerObject >( this, default_marker.Object->GetClass(), NAME_None, RF_NoFlags, default_marker.Object ) );
 }
 
 #if WITH_EDITOR
 void USFSplineComponent::SaveSplineMarkers( const TArray< FSFSplineMarker > & markers_to_save )
 {
     SplineMarkers.Reset();
 
-    const auto create_marker = [ & ]( const auto & marker, USFSplineMarkerObject * object ) {
-        object->Sprite = marker.Object->Sprite;
-        object->Color = marker.Object->Color;
-        return FSFSplineMarker( marker.Name, marker.ItIsEnabled, marker.Infos, object );
-    };
-
     for ( const auto & marker : markers_to_save )
     {
-        if ( const auto * action_marker = Cast< USFSplineMarkerObject_Action >( marker.Object ) )
-        {
-            auto new_marker = create_marker( marker, NewObject< USFSplineMarkerObject_Action >( this, marker.Object->GetClass() ) );
-            Cast< USFSplineMarkerObject_Action >( new_marker.Object )->ActionClass = action_marker->ActionClass->GetOwnerClass();
-            SplineMarkers.Emplace( new_marker );
-        }
-
-        if ( const auto * level_actor_marker = Cast< USFSplineMarkerObject_LevelActor >( marker.Object ) )
+        if ( marker.Object != nullptr )
         {
-            auto new_marker = create_marker( marker, NewObject< USFSplineMarkerObject_LevelActor >( this, marker.Object->GetClass() ) );
-            Cast< USFSplineMarkerObject_LevelActor >( new_marker.Object )->LevelActor = level_actor_marker->LevelActor;
-            SplineMarkers.Emplace( new_marker );
-        }
-
-        if ( const auto * data_marker = Cast< USFSplineMarkerObject_Data >( marker.Object ) )
-        {
-            auto new_marker = create_marker( marker, NewObject< USFSplineMarkerObject_Data >( this, marker.Object->GetClass() ) );
-            Cast< USFSplineMarkerObject_Data >( new_marker.Object )->Data = data_marker->Data;
-            SplineMarkers.Emplace( new_marker );
+            SplineMarkers.Emplace( FSFSplineMarker(
+                marker.Name,
+                marker.ItIsEnabled,
+                marker.Infos,
+                NewObject< USFSplineMarkerObject >( this, marker.Object->GetClass(), NAME_None, RF_NoFlags, marker.Object ) ) );
```

Comment:
you could use CreateMarkerFromDefault here?

Reply:
Done